rfcs/039-table-data-quality-checks.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: table_data_quality_checks
+- Start Date: 2021-07-16
+- RFC PR: [amundsen-io/rfcs#XX](https://github.com/amundsen-io/rfcs/pull/XX)
+# Table Data Quality Checks
+
+## Summary
+This RFC will define a generic interface for accessing data quality checks in Amundsen. If available, we aim to expose
+the status of these data quality checks in the Amundsen Table Details page.
+
+
+## Terms
+- Data Quality Check - Like unit tests for code, data quality checks are assertions that try to guarantee certain 
+  characteristics of data. For example a data quality check could be as simple as asserting that a column is not null,
+  or implement some complex custom logic.
```

Comment:
Do you already have in mind some tools from which these quality checks could have been collected? I can think of Great Expectations and dbt tests but would be useful to have broader list. It's also something @sewardgw had good point about the interface to be generic enough we are not forced to do backwards incompatible changes later on.

From what I can see the mockups you introduced somewhat adhere to GE format but I am not sure about dbt (cause I don't have exp with it).

Reply:
We are not building this with a particular tool in mind. At the 1M-mile view, we think our proposed definition is so generic that it could crudely fit anything. The idea is to start here and expand once people start plugging in particular systems. 

At Lyft we do have a DQ system we're planning to plug in, but it exposes a much more complicated API and semantics which we are not taking into account here. Instead, we asked ourselves what would be the most generic first step.